core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -121,6 +124,36 @@
   private SqlFunctions() {
   }
 
+  /** SQL TO_BASE64(string) function*/
+  public static String toBase64(String string) {
+    if (string == null) {
+      return null;
+    }
+    String base64 = new String(Base64.getEncoder().encode(string.getBytes(UTF_8)), UTF_8);
+    String temp = base64;
+    StringBuilder str = new StringBuilder();
```

Comment:
probably can be initialized at capacity `base64.length() + base64.length() / 76`